lib/semantics/mod-file.cc

```diff
@@ -363,56 +402,99 @@ bool ModFileReader::Read(const SourceName &modName) {
   if (!parsing.messages().empty() || !parsing.consumedWholeFile() ||
       !parseTree.has_value()) {
     errors_.push_back(
-        Error(modName, "Module file for '%s' is corrupt: %s"_err_en_US,
-            modName.ToString(), *path));
-    return false;
+        Error(name, "Module file for '%s' is corrupt: %s"_err_en_US,
+            name.ToString(), *path));
+    return nullptr;
   }
-  ResolveNames(*parseTree, parsing.cooked(), directories_);
-
-  const auto &it{Scope::globalScope.find(modName)};
-  if (it == Scope::globalScope.end()) {
-    return false;
+  Scope *parentScope;
+  if (!ancestor) {
+    // module: goes into global scope
+    parentScope = &Scope::globalScope;
+  } else {
+    // submodule: goes into parent module/submodule
+    auto *parent{GetSubmoduleParent(*parseTree)};
```

Comment:
Might be more readable without the conditional expression, instead using `if (auto *parent{...}) { parentScope = Read(...); } else { parentScope = ancestor; }`.  Maybe not.

Reply:
Yes, especially as it is really a three-way if whose purpose is to set parentScope.